File: src/main/java/com/qtech/bubbles/common/crash/ReportedException.java
package com.qtech.bubbles.common.crash;

import com.qtech.bubbles.BubbleBlaster;
import com.qtech.bubbles.core.utils.categories.StringUtils;

import java.io.PrintStream;
import java.io.PrintWriter;
import java.util.List;

public final class ReportedException extends RuntimeException {
    private final CrashReport crashReport;

    ReportedException(CrashReport crashReport) {
        this.crashReport = crashReport;
    }

    @Override
    public void printStackTrace() {
        String crashString = this.crashReport.toString();
        List<String> strings = StringUtils.splitIntoLines(crashString);
        for (String string : strings) {
            System.err.println(string);
        }
        BubbleBlaster.getInstance().shutdown();
    }

    @Override
    public void printStackTrace(PrintWriter err) {
        String crashString = this.crashReport.toString();
        List<String> strings = StringUtils.splitIntoLines(crashString);
        for (String string : strings) {
            err.println(string);
        }
        BubbleBlaster.getInstance().shutdown();
    }

    @Override
    public void printStackTrace(PrintStream err) {
        String crashString = this.crashReport.toString();
        List<String> strings = StringUtils.splitIntoLines(crashString);
        for (String string : strings) {
            err.println(string);
        }
        BubbleBlaster.getInstance().shutdown();
    }

    public CrashReport getCrashReport() {
        return crashReport;
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/crash/ReportedException.kt
package com.qtech.bubbles.common.crash

import com.qtech.bubbles.BubbleBlaster
import com.qtech.bubbles.core.utils.categories.StringUtils
import java.io.PrintStream
import java.io.PrintWriter

class ReportedException internal constructor(val crashReport: CrashReport) : RuntimeException() {
    override fun printStackTrace() {
        val crashString = crashReport.toString()
        val strings = StringUtils.splitIntoLines(crashString)
        for (string in strings) {
            System.err.println(string)
        }
        BubbleBlaster.instance.shutdown()
    }



    override fun printStackTrace(err: PrintWriter) {
        val crashString = crashReport.toString()
        val strings = StringUtils.splitIntoLines(crashString)
        for (string in strings) {
            err.println(string)
        }
        BubbleBlaster.instance.shutdown()
    }

    override fun printStackTrace(err: PrintStream) {
        val crashString = crashReport.toString()
        val strings = StringUtils.splitIntoLines(crashString)
        for (string in strings) {
            err.println(string)
        }
        BubbleBlaster.instance.shutdown()
    }
}